float32 theta_1
float32 theta_2
float32 theta_3
